content/docs/rfcs/21/README.md

```diff
@@ -0,0 +1,62 @@
+---
+slug: 21
+title: 21/WAKU2-FAULT-TOLERANT-STORE
+name: Waku v2 Fault-Tolerant Store
+status: raw
+editor: Sanaz Taheri <sanaz@status.im>
+contributors:
+---
+
+ The reliability of `13/WAKU2-STORE` protocol heavily relies on the fact that full nodes i.e., those who persist messages have high availability and uptime and do not miss any messages. 
+ If a node goes offline, then it will risk missing all the messages transmitted in the network during that time. 
+ In this specification, we provide a method that makes the store protocol resilient in presence of faulty nodes. 
+ Relying on this method,  nodes that have been offline for a time window will be able to fix the gap in their message history when getting back online. 
+ Moreover, nodes with lower availability and uptime can leverage this method to reliably provide the store protocol services as a full node.
+
+# Method description 
+ As the first step towards making the `13/WAKU2-STORE` protocol fault-tolerant, we introduce a new type of time-based query through which nodes fetch message history from each other based on their desired time window. 
+ This method operates based on the assumption that the querying node knows some other nodes in the store protocol which have been online for that targeted time window.  
+
+# Security Consideration
+
+The main security consideration to take into account while using this method is that a querying node has to reveal its offline time to the queried node. 
+This will gradually result in the extraction of the node's activity pattern which can lead to inference attacks. 
+
+# Wire Specification
+We extend the [HistoryQuery](/spec/13#payloads) protobuf message with two fields of `StartTime` and `EndTime` to signify the time range to be queried. 
```

Comment:
Field names should not be capitalised, though using snake_case is probably the preferred way to go here. See my comment on `proto` definition below.

Reply:
Right, fixed it!